sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1066,7 +1071,8 @@ class Analyzer(
           case ae: AnalysisException => s
         }
 
-      case f @ Filter(cond, child) if !f.resolved && child.resolved =>
+      case f @ Filter(cond, orgChild) if !f.resolved && orgChild.resolved =>
+        val child = EliminateBarriers(orgChild)
```

Comment:
Actually, do we need to do this? can `resolveExpressionRecursively` handle `AnalysisBarrier`?